Notepad.Extensions.Logging/NotepadLoggerProvider.cs

```diff
@@ -11,11 +11,15 @@ class NotepadLoggerProvider : ILoggerProvider
             stringBuilderPool = poolProvider.CreateStringBuilderPool();
         }
 
-        public static ILoggerProvider Instance { get; } = new NotepadLoggerProvider();
+        internal NotepadLoggerProvider(NotepadProviderOptions options) : this()
```

Comment:
Better use `IOptions<>` for this and make the logging options configurable through [the standard mechanism](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-3.1#configure-logging). You can use `LoggerProviderOptions.RegisterProviderOptions` and the `ProviderAliasAttribute` for that.

Reply:
FIxed. Now options can be configured through [configure method](https://docs.microsoft.com/ru-ru/dotnet/api/microsoft.extensions.dependencyinjection.optionsconfigurationservicecollectionextensions.configure?view=dotnet-plat-ext-3.1).